### PR TITLE
Repair LINSTOR tests to support latest driver version

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

**Review by stormi — COMMENTED** [review]
jobs.py needs to be updated so that the excluded linstor tests now run

File: tests/storage/linstor/test_linstor_sr.py [hunk 1]
```diff
@@ -2,31 +2,31 @@
...
+    """
+    Tests that do not use fixtures that setup the SR or import VMs,
+    because they precisely need to test SR creation and destruction,
+    and VM import.
+    """
 
+    def test_create_sr_without_linstor(self, host, lvm_disk):
         # This test must be the first in the series in this module
-        assert not master.binary_exists('linstor'), \
+        assert not host.file_exists('/usr/lib/python2.7/site-packages/linstor/linstorapi.py'), \
```
Comment (stormi) [inline tests/storage/linstor/test_linstor_sr.py:23]: Is there any other file we could test whose path does not depend on the version of python?

File: tests/storage/linstor/test_linstor_sr_intrapool_migration.py [hunk 2]
```diff
@@ -9,7 +9,6 @@
 # - A VM to import to the LINSTOR SR
 # And:
 # - access to XCP-ng RPM repository from hostA1
-# - a repo with the LINSTOR RPMs must be given using the command line param `--additional-repos`
 
 @pytest.mark.small_vm # run with a small VM to test the features
 @pytest.mark.big_vm # and ideally on a big VM to test it scales
```
Comment (stormi) [inline tests/storage/linstor/test_linstor_sr_intrapool_migration.py:14]: Isn't this test file missing a live *shared* migration test (see NFS tests) ?

File: lib/host.py [hunk 3]
```diff
@@ -291,6 +291,18 @@ def check_packages_available(self, packages):
         """ Check if a given package list is available in the YUM repositories. """
         return len(self.ssh(['repoquery'] + packages).splitlines()) == len(packages)
 
+    def get_available_package_versions(self, package):
+        return self.ssh(['repoquery', '--show-duplicates', package]).splitlines()
+
+    def is_package_installed(self, package):
+        try:
+            self.ssh(['yum', 'list', 'installed', package])
```
[the remark this answers is not in the document]
Comment (Wescoeur) [reply lib/host.py:299]: I missed that point, thanks.

File: tests/storage/linstor/test_linstor_sr.py (hunk 1 above)
Comment (Wescoeur) [reply tests/storage/linstor/test_linstor_sr.py:23]: We mainly depend on the LINSTOR python library in our driver, I will see if I can find another file installed by this package.

File: tests/storage/linstor/test_linstor_sr_intrapool_migration.py (hunk 2 above)
Comment (Wescoeur) [reply tests/storage/linstor/test_linstor_sr_intrapool_migration.py:14]: Nice catch, I checked the tests of few drivers to repair the LINSTOR part, but not NFS.

File: tests/storage/linstor/test_linstor_sr.py (hunk 1 above)
Comment (Wescoeur) [reply tests/storage/linstor/test_linstor_sr.py:23]: Well only python files are installed by this RPM...

Comment (stormi) [reply tests/storage/linstor/test_linstor_sr.py:23]: Ok

Comment (stormi) [reply tests/storage/linstor/test_linstor_sr.py:23]: Then maybe you can test for the presence of the rpm now that you wrote a function for this :)

**Review by stormi — APPROVED** [review]
Looks good. Before merging, we'll check it runs well on the CI hosts (physical and virtual).

@gduperrey please test on 8.2 CI hosts.  A full run on Q2/Q3, and a full run on the 3-host virtual pool too.

**Comment by gduperrey** [comment]
I don't understand what you're asking. A full run use both physicals or virtuals hosts. It depends the tests running.
Maybe we could discuss this on mattermost so i can understand what you have in mind.

**Comment by stormi** [comment]
I mean full run of linstor tests only.

**Review by stormi — COMMENTED** [review]
Let's also not forget that linstor must be "repaired" on 8.3 before we can merge this PR. Or, make them skipped on 8.3 for now.

**Pull request merged**